plugins/modules/azure_rm_recoveryservicesvault.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2020 Suyeb Ansari (@suyeb786)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = \
+    '''
+---
+module: azure_rm_recoveryservicesvault
+version_added: '1.1.0'
+short_description: Create and Update Azure Recovery Services vault
+description:
+    - Create Azure Recovery Services vault.
+    - Update Azure Recovery Services vault.
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+        required: true
+        type: str
+    recovery_service_vault_name:
+        description:
+            - The name of the Azure Recovery Service Vault.
+        required: true
+        type: str
+    location:
+        description:
+            - Azure Resource location.
+        required: true
+        type: str
+    enhanced_security_state:
+        description:
+            - Backup Policy ID present under Recovery Service Vault mentioned in recovery_vault_name field.
+        default: Enabled
+        type: str
+        choices:
+            - Enabled
+            - Disabled
+    soft_delete_feature_state:
+        description:
+            - Backup Policy ID present under Recovery Service Vault mentioned in recovery_vault_name field.
+        default: Enabled
+        type: str
+        choices:
+            - Enabled
+            - Disabled
+    state:
```

Comment:
Two states are recommended for state, I(state=present) is used for creation and update, and I(state=absent) is used for deletion. Thank you!

Reply:
Hello @Fred-sun ,
Due to security purpose, there is no official release for deleting Recovery Service Vault where all backups are resides.

Reply:
Hi @Fred-sun ,

The code is updated accordingly.

Reply:
Also FYI,

We have four options now
1. Creating/Updating Recovery service vault ->> State  ->> present
2. Updating Recovery Service Vault Backup Configuration ->> State ->> update
3. Delete Recovery Service Vault ->> State ->> absent

Updating Recovery Service and it's backup configuration both are different and we can not merge them.

Reply:
@sunshine69  As a convention, the options of state will either be "absent" or "present". Updating will be handled in the state "absent", when the expected configuration of the resource is different with the current configuration.

Reply:
Hi @haiyuazhang ,
There are two types of updates for the Recovery Service vault.
For Creating and Updating Recovery Service Vault we are already using the present state.
But,
Updating backup configuration is something different and we can not merge these two independent operations.

with only the present state we can not say that the user wants to update "Recovery Service Vault" or "Recovery Service Vault Backup Configuration".
Also, the API for both operations is different.
Let me know if you want me to remove the "Recovery Service Vault Backup Configuration" feature from this module.

Reply:
Hi @Fred-sun @haiyuazhang ,
Any feedback on this?

Reply:
Hi @suyeb786 Generally speaking, state is combined with state "present" and "absent". When state=present, it can judge whether the resource exists in the class or function. If it does, it can judge whether it is updated or not(idempotent or not), and if it is not, it can be created. Thank you very much!

Reply:
@Fred-sun, Thank you for the details this makes sense to me.
Updating Code accordingly.